src/components/Card.js

```diff
@@ -49,17 +89,44 @@ export class Card {
     this._deleteButton = this._cardElement.querySelector(
       ".card__delete-button"
     );
-    this._deleteButton.addEventListener("click", () => {
-      this._deleteCard();
-    });
+    this._deleteButton.addEventListener("click", this._handleDelete);
   }
 
-  _deleteCard() {
+  deleteCard() {
+    console.log("you pressed this delete button");
+    // this._handleDelete.setEventListeners();
+    // this._handleDelete.open();
     this._cardElement.remove();
     this._cardElement = null;
   }
 
+  refreshLikesCount(likesCount) {
+    likesCount < 1
+      ? (this._cardElement.querySelector(".card__likes-counter").textContent =
```

Comment:
`this._cardElement.querySelector(".card__likes-counter")`


**Could be improved**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` and so on.